src/components/pastAnalysis/PastAnalysisTable.tsx

```diff
@@ -152,17 +164,55 @@ const PastAnalysisTable = () => {
   }, [tableState, perpage, dispatch]);
 
   // Polls ChRIS backend and refreshes table if any of the plugins with the given IDs have a terminated status
+
   useInterval(async () => {
-    if (tableState.processingPluginIds) {
-      for (const id of tableState.processingPluginIds) {
-        const refresh = await ChrisIntegration.checkIfPluginTerminated(id);
-        if (refresh) {
-          // Right before updating max feed ID and refreshing table, get a list of all the "Analysis Created" properties on page 0
-          newRowsRef.current = tableState.storedPages[0]?.map((study: StudyInstanceWithSeries) => study.analysisCreated);
-          updateMaxFeedId();
-          return;
-        }
+    let finishedPlugins: number[] = [];
+
+    for (const id of tableState.processingPluginIds) {
+      if (await ChrisIntegration.checkIfPluginTerminated(id)) { // parallel async execution here
+        // Right before updating max feed ID and refreshing table, get a list of all the "Analysis Created" properties on page 0
+
+        // newRowsRef.current = tableState.storedPages[0]?.map((study: StudyInstanceWithSeries) => study.analysisCreated);
+        finishedPlugins.push(id);
+      }
+    }
```

Comment:
It should be possible to `checkIfPluginTerminated` asynchronously instead of awaiting for every single response and then adding the pluginID into the finishedPlugins. I suggest using `Promise.all()` along with `tableState.processingPluginIds.map()` to parallelize the calls.